File: VB/DXApplication36/CustomRichEditCommandFactoryService.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Windows
Imports DevExpress.XtraRichEdit
Imports DevExpress.Xpf.RichEdit
Imports DevExpress.XtraRichEdit.Services
Imports DevExpress.XtraRichEdit.Commands
Imports DevExpress.Utils

Namespace DXApplication36

    #Region "#CustomRichEditCommandFactoryService"
    Public Class CustomRichEditCommandFactoryService
        Implements IRichEditCommandFactoryService

        Private ReadOnly service As IRichEditCommandFactoryService
        Private ReadOnly control As RichEditControl

        Public Sub New(ByVal control As RichEditControl, ByVal service As IRichEditCommandFactoryService)
            Guard.ArgumentNotNull(control, "control")
            Guard.ArgumentNotNull(service, "service")
            Me.control = control
            Me.service = service
        End Sub

        Public Function CreateCommand(ByVal id As RichEditCommandId) As RichEditCommand Implements IRichEditCommandFactoryService.CreateCommand
            If id = RichEditCommandId.FileSave Then
                Return New CustomSaveDocumentCommand(control)
            End If

            Return service.CreateCommand(id)
        End Function
    End Class
    #End Region ' #CustomRichEditCommandFactoryService

    #Region "#customsavedocumentcommand"
    Public Class CustomSaveDocumentCommand
        Inherits SaveDocumentCommand

        Public Sub New(ByVal control As IRichEditControl)
            MyBase.New(control)
        End Sub
        Protected Overrides Sub ExecuteCore()
              MyBase.ExecuteCore()
              MessageBox.Show("Custom SaveDocument command was executed")
        End Sub
    End Class
    #End Region ' #customsavedocumentcommand
End Namespace

File: CS/DXApplication36/CustomRichEditCommandFactoryService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Windows;
using DevExpress.XtraRichEdit;
using DevExpress.Xpf.RichEdit;
using DevExpress.XtraRichEdit.Services;
using DevExpress.XtraRichEdit.Commands;
using DevExpress.Utils;

namespace DXApplication36 {
    public class CustomRichEditCommandFactoryService : IRichEditCommandFactoryService {
        readonly IRichEditCommandFactoryService service;
        readonly RichEditControl control;

        public CustomRichEditCommandFactoryService(RichEditControl control,
            IRichEditCommandFactoryService service) {
            Guard.ArgumentNotNull(control, "control");
            Guard.ArgumentNotNull(service, "service");
            this.control = control;
            this.service = service;
        }

        public RichEditCommand CreateCommand(RichEditCommandId id) {
            if(id == RichEditCommandId.FileSave)
                return new CustomSaveDocumentCommand(control);

            return service.CreateCommand(id);
        }
    }
   
    public class CustomSaveDocumentCommand:SaveDocumentCommand{
        public CustomSaveDocumentCommand(IRichEditControl control)
            : base(control)
        {
        }
        protected override void ExecuteCore()
        {
              base.ExecuteCore();
              MessageBox.Show("Custom SaveDocument command was executed");
        }
    }
}
